plaid/plaid.go

```diff
@@ -1,373 +1,136 @@
 	"errors"
+	"fmt"
 	"io"
 	"io/ioutil"
 	"net/http"
-)
+	"path/filepath"
+	"strings"
 
-// NewClient instantiates a Client associated with a client id, secret and environment.
-// See https://plaid.com/docs/api/#gaining-access.
-func NewClient(clientID, secret string, environment environmentURL) *Client {
-	return &Client{clientID, secret, environment, &http.Client{}}
-}
+	"github.com/BurntSushi/toml"
+)
 
-// Same as above but with additional parameter to pass http.Client. This is required
-// if you want to run the code on Google AppEngine which prohibits use of http.DefaultClient
-func NewCustomClient(clientID, secret string, environment environmentURL, httpClient *http.Client) *Client {
-	return &Client{clientID, secret, environment, httpClient}
-}
+// PlaidAPIVersion holds the latest version of the Plaid API
+const PlaidAPIVersion = "2018-05-22"
 
+// Client holds information required to interact with the Plaid API.
 // Note: Client is only exported for method documentation purposes.
 // Instances should only be created through the 'NewClient' function.
 //
 // See https://github.com/golang/go/issues/7823.
 type Client struct {
-	clientID    string
-	secret      string
-	environment environmentURL
-	httpClient  *http.Client
-}
-
-type environmentURL string
-
-var Tartan environmentURL = "https://tartan.plaid.com"
-var Production environmentURL = "https://api.plaid.com"
-
-type Account struct {
-	ID      string `json:"_id"`
-	ItemID  string `json:"_item"`
-	UserID  string `json:"_user"`
-	Balance struct {
-		Available float64 `json:"available"`
-		Current   float64 `json:"current"`
-	} `json:"balance"`
-	Meta struct {
-		Number string `json:"number"`
-		Name   string `json:"name"`
-	} `json:"meta"`
-	Numbers struct {
-		Account     string `json:"account"`
-		Routing     string `json:"routing"`
-		WireRouting string `json:"wireRouting"`
-	} `json:"numbers"`
-	Type            string `json:"type"`
-	InstitutionType string `json:"institution_type"`
-}
-
-type Transaction struct {
-	ID        string `json:"_id"`
-	AccountID string `json:"_account"`
-
-	Amount float64 `json:"amount"`
-	Date   string  `json:"date"`
-	Name   string  `json:"name"`
-	Meta   struct {
-		AccountOwner string `json:"account_owner"`
-
-		Location struct {
-			Address string `json:"address"`
-			City    string `json:"city"`
-
-			Coordinates struct {
-				Lat float64 `json:"lat"`
-				Lon float64 `json:"lon"`
-			} `json:"coordinates"`
-
-			State string `json:"state"`
-			Zip   string `json:"zip"`
-		} `json:"location"`
-	} `json:"meta"`
-
-	Pending bool `json:"pending"`
-
-	Type struct {
-		Primary string `json:"primary"`
-	} `json:"type"`
-
-	Category   []string `json:"category"`
-	CategoryID string   `json:"category_id"`
-
-	Score struct {
-		Location struct {
-			Address float64 `json:"address"`
-			City    float64 `json:"city"`
-			State   float64 `json:"state"`
-			Zip     float64 `json:"zip"`
-		}
-		Name float64 `json:"name"`
-	} `json:"score"`
+	ClientID    string
```

Comment:
we should not export these fields -- `Client` should be opaque to the caller.

Reply:
⚡️